### ISSUE TITLE
joblib.Parallel overwrites args variable for n_jobs > 1

### ISSUE THREAD

**Issue by mgbckr** [joblib/joblib#833]
joblib version: 0.13.0

In the following example, when I set `n_jobs>1`, then the args from the outer scope will be overwritten by a Paralel namespace for  `n_jobs=1` this does not happen. Other variable names are not effected. This a) is not consistent and b) generally, I think Parallel should not overwrite _any_ variable. Thanks :)

```python
import joblib

args=5
def func():
    return args

joblib.Parallel(n_jobs=2)(joblib.delayed(func)() for i in range(1))
```

**Update:**

I am using Ubuntu:
```bash
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

Kernel:
```
Linux 4.15.0-43-generic x86_64
```

**Comment by mgbckr** [comment]
Hi, I added my OS details. I think adding this in a visible way to the documentation of `joblib.Parallel` (and maybe `joblib.delayed`?) would be fine.

**Comment by ogrisel** [comment]
I confirm that the fix in cloudpipe/cloudpickle#240 fixes this issue.
